converter/internal/prometheusconvert/prometheusconvert.go

```diff
@@ -106,34 +106,34 @@ func appendServiceDiscoveryConfigs(pb *prometheusBlocks, serviceDiscoveryConfig
 			targets = append(targets, getScrapeTargets(sdc)...)
 		case *prom_azure.SDConfig:
 			labelCounts["azure"]++
-			exports = appendDiscoveryAzure(pb, common.GetUniqueLabel(label, labelCounts["azure"]), sdc)
+			exports = appendDiscoveryAzure(pb, common.GetLabelWithIndex(labelCounts["azure"]-1, label), sdc)
 		case *prom_consul.SDConfig:
 			labelCounts["consul"]++
-			exports = appendDiscoveryConsul(pb, common.GetUniqueLabel(label, labelCounts["consul"]), sdc)
+			exports = appendDiscoveryConsul(pb, common.GetLabelWithIndex(labelCounts["consul"]-1, label), sdc)
```

Comment:
nit: This looks odd with the `++` followed by an immediate `-1`, can we reverse the order and not have to do the -1?

Reply:
I don't love this either but it saves a line of code for each. I think this code will be refactored and disappear once the dust settles so I'm inclined to leave it for today.

longer explanation:
`labelCounts["consul"]++` will:
- add `consul` to the slice with an initial value of 1
- increment `consul`

Since this is 1 indexed it creates the unfortunate offset.